pkg/sectigo/server/server.go

```diff
@@ -0,0 +1,217 @@
+/*
+Package server implements a lightweight Sectigo mock server that can be used in staging
+to issue mock certificates and perform integration tests. This server implements a
+subset of the Sectigo IoT API that targets GDS-specific usage. All state is held
+in-memory and is periodically flushed so this service should not be relied on for
+anything other than staging and systems integration tests.
+*/
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg"
+	"github.com/trisacrypto/directory/pkg/utils/logger"
+)
+
+func init() {
+	// Initialize zerolog with GCP logging requirements
+	zerolog.TimeFieldFormat = time.RFC3339
+	zerolog.TimestampFieldName = logger.GCPFieldKeyTime
+	zerolog.MessageFieldName = logger.GCPFieldKeyMsg
+
+	// Add the severity hook for GCP logging
+	var gcpHook logger.SeverityHook
+	log.Logger = zerolog.New(os.Stdout).Hook(gcpHook).With().Timestamp().Logger()
+}
+
+type Server struct {
```

Comment:
Really just our standard gin api server but really light on the impelmentation, logging, o11y, etc. This is not a production service and is intended solely for staging integration testing.